tools/sdk-generation-pipeline/pipelines/steps/parse-and-publish-result.yml

```diff
@@ -0,0 +1,129 @@
+          --resultOutputPath="/tmp/output/${TASK_NAME}-task-result.json" \
+          --dockerResultFile="/tmp/output/taskResults.json" \
+          --taskOutputPath="${TASK_OUTPUT_PATH}"
+        retVal=$?
+        if [ $retVal -ne 0 ]; then
+            #TODO: when all language support mock test, set pipelineResult for parse task result
+            echo "wait all language support mock test"
+            #echo "##vso[task.setvariable variable=pipelineResult;]failed"
+        fi
+      done
+    displayName: Parse task result
+    condition: or(contains(variables['pipelineResult'], 'success'), contains(variables['pipelineResult'], 'failed'))
+
+  - bash: |
+      prepareArtifactFiles \
+        --artifactDir="$(System.ArtifactsDirectory)" \
+        --generateAndBuildOutputFile="/tmp/output/generateAndBuildOutputJson.json" \
+        --language="${{ parameters.sdk }}"
+      retVal=$?
+      if [ $retVal -ne 0 ]; then
+          echo "##vso[task.setvariable variable=pipelineResult;]failed"
+          exit 1
+      fi
+      cp /tmp/output/docker.log $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-full.log
+      cp /tmp/output/init-task.log $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-init.log
+      cp /tmp/output/init-task-result.json $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-init-result.json
+      cp /tmp/output/generateAndBuild-task.log $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-generateAndBuild.log
+      cp /tmp/output/generateAndBuild-task-result.json $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-generateAndBuild-result.json
+      cp /tmp/output/mockTest-task.log $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-mockTest.log | echo "wait all language support mock test"
+      cp /tmp/output/mockTest-task-result.json $(System.ArtifactsDirectory)/logs/${{ parameters.sdkGenerationName }}-mockTest-result.json | echo "wait all language support mock test"
+    displayName: Prepare artifact
+    workingDirectory: $(SDK_REPO)
+    condition: contains(variables['pipelineResult'], 'success')
+
+  - task: PublishPipelineArtifact@1
+    displayName: Publish SDK Generation pipeline artifact
+    inputs:
+      targetPath: $(System.ArtifactsDirectory)/
+      artifactName: sdk-generation
+    condition: contains(variables['pipelineResult'], 'success')
+
+  - bash: |
+      publishResult \
+        --storageType=eventhub \
+        --pipelineStatus=completed \
+        --buildId="$(Build.BuildId)" \
+        --trigger='${{ parameters.trigger }}' \
+        --logPath=$(Build.BuildId)/logs \
+        --resultsPath='["/tmp/output/init-task-result.json",
+          "/tmp/output/generateAndBuild-task-result.json",
+          "/tmp/output/mockTest-task-result.json"]'
+      retVal=$?
+      if [ $retVal -ne 0 ]; then
+          echo "##vso[task.setvariable variable=pipelineResult;]failed"
```

Comment:
Why are you setting this variable explicitly? If you exit 1 then devops will mark this pipeline as failed.

Reply:
thx for the suggestion, simplify judgment conditions